### PR TITLE
Show that a Post/Comment has been edited

### LINKED ISSUE #1669: 🚀 [Feature] Show "Edited" on comments and Posts
> [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-22T22:42:38Z" title="Monday, September 23rd 2019, 12:42:38 am +02:00">Sep 23, 2019</time>_
_Closed <time datetime="2019-10-04T12:18:44Z" title="Friday, October 4th 2019, 2:18:44 pm +02:00">Oct 4, 2019</time>_
---

## :rocket: Feature

A user has pointed out that even if we show placeholders for deleted comments people could still distort replies to a post or a comment by editing the content of a post or comment.

As a first step, we should compare `createdAt` with `updatedAt` and show a hint "Edited" along with the date of the edit. That way, we give readers at least the option to see this, even if they have to look carefully to spot it.

By the way, Github shows a full edit history for each comment.

### Design & Layout

Screenshot - 2019-09-23T004325 311

### Additional context
Here is the post where I got the inspiration: https://alpha.human-connection.org/post/54530e00-032a-4221-a225-9d9caf2a5d3c/welchen-sinn

### PR DESCRIPTION
**Pull request by Tirokk**
> [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-10-03T10:02:51Z" title="Thursday, October 3rd 2019, 12:02:51 pm +02:00">Oct 3, 2019</time>_
_Merged <time datetime="2019-10-04T12:18:44Z" title="Friday, October 4th 2019, 2:18:44 pm +02:00">Oct 4, 2019</time>_
---

### Issues

- fixes #1669

### CODE REVIEW

File: webapp/components/User/User.vue [hunk 1]
```diff
@@ -11,17 +11,17 @@
...
           <div>
-            <ds-text align="left">
+            <ds-text>
               <b class="username">{{ userName | truncate(18) }}</b>
               <ds-text v-if="dateTime" size="small" color="soft">
                 <ds-icon name="clock" />
                 <client-only>
                   <hc-relative-date-time :date-time="dateTime" />
                 </client-only>
+                <slot name="dateTime"></slot>
```
Comment (Tirokk) [inline webapp/components/User/User.vue:21]: > [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-10-04T10:06:56Z" title="Friday, October 4th 2019, 12:06:56 pm +02:00">Oct 4, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -11,17 +11,17 @@
         <div @mouseover="openMenu(true)" @mouseleave="closeMenu(true)">
           <hc-avatar class="avatar" :user="user" />
           <div>
-            <ds-text align="left">
+            <ds-text>
               <b class="username">{{ userName | truncate(18) }}</b>
               <ds-text v-if="dateTime" size="small" color="soft">
                 <ds-icon name="clock" />
                 <client-only>
                   <hc-relative-date-time :date-time="dateTime" />
                 </client-only>
+                <slot name="dateTime"></slot>
```


why is this slot called `dateTime`??

File: webapp/pages/post/_id/_slug/index.vue [hunk 2]
```diff
@@ -6,8 +6,11 @@
       :class="{ 'post-card': true, 'disabled-content': post.disabled }"
     >
       <ds-space margin-bottom="small" />
-      <hc-user :user="post.author" :date-time="post.createdAt" />
-      <!-- Content Menu (can open Modals) -->
+      <hc-user :user="post.author" :date-time="post.createdAt">
+        <template v-slot:dateTime>
```
Comment (Tirokk) [inline webapp/pages/post/_id/_slug/index.vue:10]: > [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-10-04T10:07:11Z" title="Friday, October 4th 2019, 12:07:11 pm +02:00">Oct 4, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -6,8 +6,11 @@
       :class="{ 'post-card': true, 'disabled-content': post.disabled }"
     >
       <ds-space margin-bottom="small" />
-      <hc-user :user="post.author" :date-time="post.createdAt" />
-      <!-- Content Menu (can open Modals) -->
+      <hc-user :user="post.author" :date-time="post.createdAt">
+        <template v-slot:dateTime>
```


same as above :point_up_2:

File: webapp/components/User/User.vue (hunk 1 above)
Comment (Tirokk) [reply webapp/components/User/User.vue:21]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-10-04T11:30:03Z" title="Friday, October 4th 2019, 1:30:03 pm +02:00">Oct 4, 2019</time>_

---


My thought was to add additional data to dateTime. Therefore I called the named slot `dateTime`.

**Review by Mogge — APPROVED** [review]
edited

@mattwr18 I edited this PR, have a look! Feel free to merge this, if you want.

**Pull request closed without merging**